invokeai/app/invocations/baseinvocation.py

```diff
@@ -720,21 +800,30 @@ def wrapper(cls: Type[TBaseInvocation]) -> Type[TBaseInvocation]:
         validate_fields(cls.model_fields, invocation_type)
 
         # Add OpenAPI schema extras
-        uiconf_name = cls.__qualname__ + ".UIConfig"
-        if not hasattr(cls, "UIConfig") or cls.UIConfig.__qualname__ != uiconf_name:
-            cls.UIConfig = type(uiconf_name, (UIConfigBase,), {})
-        if title is not None:
-            cls.UIConfig.title = title
-        if tags is not None:
-            cls.UIConfig.tags = tags
-        if category is not None:
-            cls.UIConfig.category = category
+        uiconfig_name = cls.__qualname__ + ".UIConfig"
+        if not hasattr(cls, "UIConfig") or cls.UIConfig.__qualname__ != uiconfig_name:
+            cls.UIConfig = type(uiconfig_name, (UIConfigBase,), {})
+        cls.UIConfig.title = title
+        cls.UIConfig.tags = tags
+        cls.UIConfig.category = category
+
+        # Grab the node pack's name from the module name, if it's a custom node
+        module_name = cls.__module__.split(".")[0]
+        if module_name.endswith(CUSTOM_NODE_PACK_SUFFIX):
+            cls.UIConfig.node_pack = module_name.split(CUSTOM_NODE_PACK_SUFFIX)[0]
+        else:
+            cls.UIConfig.node_pack = None
```

Comment:
We could do something like this:
```py
is_custom = cls.__module__.rsplit(".", 1)[0] == 'invokeai.app.invocations'
```

If a custom node structured its modules correctly, it could impersonate a core node.

Probably the better way to do this is to have `class CustomInvocation(BaseInvocation)` and `class CoreInvocation(BaseInvocation)`, and make it hard for custom nodes to import `CoreInvocation`. This is predicated on splitting the invocation registration logic from `BaseInvocation`.

The goal here isn't necessarily to prevent nodes from doing something malicious - its more to be able to differentiate core nodes from custom nodes in the interest of providing a good user and developer experience.

Reply:
I wasn't even thinking from a security perspective - just seemed a little fragile. Let's not bother with trying to solve this right now. It's easy to change in the future. And then we can take the time to fully consider how we want to handle core vs. community.